Tools/GNUMake/sites/Make.unknown

```diff
@@ -20,101 +20,34 @@ ifeq ($(USE_MPI),TRUE)
-
-    #
-    # mvapich
-    #
-
-#    ifneq ($(BL_NO_FORT),TRUE)
-      mpif90_link_flags := $(shell $(MPI_OTHER_COMP) -link_info)
-      #
-      # The first word is the underlying compiler say gfortran
-      #
-      LIBRARIES += $(wordlist 2,1024,$(mpif90_link_flags))
-#    endif
-
-#    ifneq ($(CXX),$(filter $(CXX),mpicxx mpic++))
-      mpicxx_link_flags := $(shell mpicxx -link_info)
-      LIBRARIES += $(filter -l%,$(mpicxx_link_flags))
-#    endif
-
-  else ifneq ($(findstring Open MPI, $(shell $(MPI_OTHER_COMP) -showme:version 2>&1)),)
-
-    #
-    # openmpi
-    #
-
-#    ifneq ($(BL_NO_FORT),TRUE)
-      mpif90_link_flags := $(shell $(MPI_OTHER_COMP) -showme:link)
-      LIBRARIES += $(mpif90_link_flags)
-#    endif
-
-#    ifneq ($(CXX),$(filter $(CXX),mpicxx mpic++))
-      mpicxx_link_flags := $(shell mpicxx -showme:link)
-      LIBRARIES += $(filter -l%,$(mpicxx_link_flags))
-#    endif
-
-     DEFINES += -DOMPI_SKIP_MPICXX
-
-  else ifneq ($(findstring Spectrum MPI, $(shell $(MPI_OTHER_COMP) -showme:version 2>&1)),)
-
-    #
-    # Spectrum MPI
-    #
-#    ifneq ($(BL_NO_FORT),TRUE)
-      mpif90_link_flags := $(shell $(MPI_OTHER_COMP) -showme:link)
-      LIBRARIES += $(mpif90_link_flags)
-#    endif
-
-  else ifneq ($(findstring /intel/mpi-rt,$(shell $(MPI_OTHER_COMP) -show 2>&1)),)
-
-    #
-    # Intel mpi
-    #
-#
-#    ifneq ($(BL_NO_FORT),TRUE)
-      mpif90_link_flags := $(shell $(MPI_OTHER_COMP) -link_info)
-      #
-      # The first word is the underlying compiler say gfortran
-      #
-      LIBRARIES += $(wordlist 2,1024,$(mpif90_link_flags))
-#    endif
-
-#    ifneq ($(CXX),$(filter $(CXX),mpicxx mpic++))
-      mpicxx_link_flags := $(shell mpicxx -link_info)
-      LIBRARIES += $(filter -l%,$(mpicxx_link_flags))
-#    endif
-
+  # error codes - find flag that displays the data
+  code_sml := $(shell $(MPI_OTHER_COMP) -showme:link > /dev/null 2>&1; echo $$?)
+  code_li :=  $(shell $(MPI_OTHER_COMP) -link_info   > /dev/null 2>&1; echo $$?)
+
+  # f90: filter all flags (-%)
+  # cxx: filter additional library flags (-l%)
+  ifeq ($(code_sml), 0)
+     mpif90_link_flags := $(filter -%,$(shell $(MPI_OTHER_COMP) -showme:link))
+     mpicxx_link_flags := $(filter -l%,$(shell mpicxx -showme:link))
+  else ifeq ($(code_li), 0)
+     mpif90_link_flags := $(filter -%,$(shell $(MPI_OTHER_COMP) -link_info))
+     mpicxx_link_flags := $(filter -l%,$(shell mpicxx -link_info))
   else
+     $(error Unknown mpi wrapper.  You can try setting MPI stuff in amrex/Tools/GNUMake/Make.local and then compile with NO_MPI_CHECKING=TRUE.)
+  endif
 
-    $(error Unknown mpi implementation.  You can try setting MPI stuff in amrex/Tools/GNUMake/Make.local and then compile with NO_MPI_CHECKING=TRUE.)
+  LIBRARIES += $(mpif90_link_flags) $(mpicxx_link_flags)
 
+  # OpenMPI specific flag
+  ifneq ($(findstring Open MPI, $(shell $(MPI_OTHER_COMP) -showme:version 2>&1)),)
+     DEFINES += -DOMPI_SKIP_MPICXX
```

Comment:
It's probably safe to always define this.

Reply:
👍 I'll comment this out for now (and future reference) and we can test to make sure it doesn't break anything.